src/tool/cppwinrt/strings/base_coroutine_foundation.h

```diff
@@ -666,9 +666,36 @@ namespace std::experimental
 
 namespace winrt
 {
-    template <typename... Args>
-    Windows::Foundation::IAsyncAction when_all(Args&&... args)
+    template <typename... T>
+    Windows::Foundation::IAsyncAction when_all(T... async)
+    {
+        (co_await async, ...);
+    }
+
+    template <typename... T>
+    Windows::Foundation::IAsyncAction when_any(T&& ... async)
     {
-        (co_await args, ...);
+        static_assert((impl::has_category_v<std::decay_t<T>> && ...), "Async must be WinRT async type such as IAsyncAction or IAsyncOperation<T>.");
+
+        handle event{ check_pointer(WINRT_CreateEventW(nullptr, true, false, nullptr)) };
+        void* process = WINRT_GetCurrentProcess();
+
+        auto duplicate = [&](handle const& source) noexcept
+        {
+            void* target{};
+            WINRT_VERIFY(WINRT_DuplicateHandle(process, source.get(), process, &target, 0, false, 2 /* DUPLICATE_SAME_ACCESS */));
+            return handle{ target };
+        };
+
+        auto completed = [&](auto&& async)
```

Comment:
This "completed" lambda isn't technically needed, but it is here to [work around a compiler bug](https://developercommunity.visualstudio.com/content/problem/648507/c-pack-expansion-crashes-compile-with-lambda-expre.html).